stock_cancel/models/stock.py

```diff
@@ -46,16 +47,32 @@ def action_revert_done(self):
                     % (picking.name))
             if picking.invoice_id:
                 raise exceptions.Warning(
-                    _('Picking %s has invoices!') % (picking.name))
+                    _('Picking %s has been invoiced') % (picking.name))
+            for move in picking.move_lines:
+                quants = move.quant_ids.filtered(
```

Comment:
Also it won't work in multi-location databases, where the actual location of the quant is decided using pack operations not stock moves (using manual entry and automatic put-away strategy).
Usually you have:
- move dest = the warehouse's location
- pack operation dest = the actual place where the goods were put.